examples/basic/topology_optimization_cantilever.py

```diff
@@ -0,0 +1,188 @@
+""" .. _ref_topology_optimization:
+
+Topology Optimization of a Simple Cantilever
```

Comment:
@dipinknair can we add "Beam" at the end, like:
"Topology Optimization of a Simple Cantilever Beam"